### ISSUE TITLE
See CO2 estimation at serviceId level

### ISSUE THREAD

**Issue by lecharcutier** [cloud-carbon-footprint/cloud-carbon-footprint#625]
Feature Suggestion
I try to use cloud carbon footprint to estimate the consumption of each service of my apps.
My azure accounts are organized heterogeneously. Sometimes, I have one environment per account, sometimes several environments per account with a resource group segregation.
Anyway, in order to be able to estimation the carbon consumption for each service of each app, I would like to be able to retrieve this consumption at "resource ID" level
 
**Describe the solution you'd like**
I would like to be able to have the following result on the CCF API : 
{date; Cloud Provider; Account; region; service name; **resource group; serviceID**; kilowattHours;CO2e}
This need may be extended to AWS for sure!
 
**Describe alternatives you've considered**
Having at least "Service ID" would be enough, since it is possible to retrieve the Ressource group name with Azure API call afterward.
 
## Possible Implementation
No idea sorry :) but I do know that the Azure bill API provides this level of detail.

**Comment by ericksod** [comment]
Hi @lecharcutier!
Thanks for your suggestion - just wondering, would having the serviceId included in the API be enough, or are you looking to see something about this in the dashboard?

**Comment by lecharcutier** [comment]
Hello @ericksod,

Thanks a lot for this response ! 
ServiceID will be enough in the API since it's quite easy to retrieve the Ressource Group of a resourceID afterward. 

About the dashboard, I don't think it is usefull to see such detailed level. On my side, the idea  is to be able to use the API to populate BI report with carbon Data (and in those reports, ressourceID is a key).

Thanks again!

**Comment by ericksod** [comment]
Thanks for clarification @lecharcutier!
We'll add a ticket for this to our backlog, and perhaps a contributor could look into it.

**Comment by ericksod** [comment]
Hi @lecharcutier! After discussing with the team, adding ServiceID doesn't seem like it would be straightforward with the current API design and we anticipate it to be a significant effort.
Out of curiosity for your use case, are you able to give further detail on what insights or analysis you'd hope to do with the resourceID and subsequent BI report?

**Comment by lecharcutier** [comment]
Hello sorry for the late response; 
i work for several large firm and they have different type / way of organizing their Azure subscriptions. 
Sometimes, one subscription contain many application (segregated with RG), some time one subscription contain one app, or often they decide to dedication one subscription for prod env and one for non prod.
Anyway, my objectif are the following  : First, i'de like to be able to use CCF to estimation the CO2 consumption per app. In the Case of multi-app hosting (one subscription for several app), CCF don't allow to do that at the moment without the suggested feature

secondly, (more long term approach), i would like to be able to traduce my FinOps optimisation into CO2 consumption decrease. This could help me to communication on anti-waste principle per services. The idea is mostly to contest € in CO2 in order to motivate teams to take care of their cloud consumption.

hope it help, I'll be glad to give you more detail if needed

**Comment by tpatrizio** [comment]
Hello All, I have exactly the same requirement of @lecharcutier, I'd like to leverage CFP APIs to feed a PowerBI dashboard that offer the possibility to filter down data at resource group level. It could also be acceptable to leverage resource tagging to evaluate the carbon frootprint of just the resources used by a given project, is this someway possible?

**Comment by 4upz** [comment]
Hi all, thanks for the suggestion and conversation around this! We've began exploring including tagging for each cloud provider, which should fit the use case of drilling down to a resource level as specified in the original description. So far, we've added tagging support for AWS and are looking into the other providers next. For Azure, we'll begin exploring and tracking the progress within this new issue https://github.com/cloud-carbon-footprint/cloud-carbon-footprint/issues/964 but if anyone wants to begin taking a stab at this, feel free to let us know! Closing this issue since the feature will be implemented with that ticket
